### PR TITLE
Implement basic mob AI from pureentities

### PR DESCRIPTION
**Pull request by Creeperface01**

### CODE REVIEW

**Comment by Pub4Game** [comment]
It is implemented very badly, even Genisys deleted it

File: src/main/java/cn/nukkit/level/generator/biome/Biome.java [hunk 1]
```diff
@@ -171,5 +175,18 @@ private static int generateBiomeColor(double temperature, double rainfall) {
         return new double[]{a[0] * invs + b[0] * s, a[1] * invs + b[1] * s, a[2] * invs + b[2] * s};
     }
 
+    public int getStoneBlock() {
+        return this.stoneBlock;
+    }
+
+    public int getSurfaceBlock() {
```
Comment (Pub4Game) [inline src/main/java/cn/nukkit/level/generator/biome/Biome.java]: What is it? Oh God no

**Comment by Creeperface01** [comment]
Idk but caves look good :D
screenshot_2016-10-05-19-46-52
screenshot_2016-10-05-19-47-27
screenshot_2016-10-05-19-47-40

**Comment by Creeperface01** [comment]
And what should be changed? Remove Moving entity and implement it into Entity? But i think we should distinguish flying entity and walking

**Comment by Pub4Game** [comment]
I asked to split the pull request... ONE pull request - one fix/enhancement

**Comment by Creeperface01** [comment]
OK and just what's wrong and any idea how AI system should look like?

**Comment by Pub4Game** [comment]
This AI is very flawed and I think it is better to leave it as a plugin

**Comment by Muqsit** [comment]
The plugin does have a really smooth ai tho. Maybe a switch for ai in nukkit.yml might do? Idk..

**Comment by Creeperface01** [comment]
i know it needs some improvements but it will be good :) nobody will make AI that will be like orginal MC AI....

**Comment by Snake1999** [comment]
Entity AI should all be internal classes, which shouldn't be public. As it become internal, I think a new AI package shouldn't be create. Just place them into cn.nukkit.entity .
And push 2 different improvements into 2 different PRs.

**Comment by Snake1999** [comment]
Okay, merge this first then... :P

**Comment by Pub4Game** [comment]
No... god... please no....

**Comment by Muqsit** [comment]
Aww...you could've fixed it.. that AI works really smooth in Nukkit.

**Pull request merged**